### ISSUE TITLE
Separate spaces Error 403 (you are not allowed to join this room)

### ISSUE THREAD

**Issue by hellkaim** [mautrix/discord#67]
Hello and thank you for awesome work! 
Being using you're TG and WA bridges for a while (actually couple of years) and decided to give it a go for Discord's one.

While the bot seems to work (it can see my guilds and connects me to 1-1 chats there is an issue with guilds with multiple channels.

So I issue a command 'guild bridge ID' and the bot starts to populate rooms.

The Matrix reports back with "can't ensure joining to the room" and "MatrixError 403: You are not allowed to join this room".

From the Element client (both android and web) the error is the same. I can see rooms are created, but I can't join them.

Question: what should I check first if this is indeed an error and not intentional behaviour
!
Attachin literary a screenshot ))
[IMG_20230306_225818_4](https://user-images.githubusercontent.com/40304971/223276157-9e018e49-31e6-43c0-a41f-0527872063f3.jpg)
?

**Comment by tulir** [comment]
You'd need to look at the bridge or server logs. If it's dendrite then it's probably a server bug as it has known issues with state handling.

**Comment by SNThrailkill** [comment]
@hellkaim did you ever get this figured out?

**Comment by hellkaim** [comment]
@SNThrailkill  Partially. In my case it was due to the bug in Synapse release. Downgrading a couple of minor releases did the trick. It was due to some wired staff with encryption deep in the code.

**Comment by SNThrailkill** [comment]
@hellkaim thank you for confirming. Looks like the bug is partially back in 1.89.0. I'll keep an eye on this.

**Comment by hellkaim** [comment]
Sad to hear that. 
Unfortunately I did not bother to make any notes so I can't really point you anywhere, but I remember that the issue was in how Synapse was dealing with encryption tokens or something like that.
I remember this, that it can be seen in the main server log as 1 line complaining regarding either a hand shake or access restriction, and it led me to kind of a rabbit hole where I stumbled across several Git and Reddit threads all discussing encryption issues. It was way complicated then what I can digest so I went with downgrade.

To be fair, the issue dissapiared after about tow minor releases (both up or down from the release I was on). I waited about 2 month and then rolled to the current stable that prove to be working fine.

I am currently on 1.83.0
